components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthUtil.java

```diff
@@ -830,4 +836,30 @@ public static String getUsername(String tenantDomain) throws IdentityApplication
         }
         return username;
     }
+
+    /**
+     * Resolves user id from username in scenarios where user id is set as the username in organization specific flows
+     *
+     * @param authorizedUser authorized user.
+     * @return userId  The user id.
+     */
+    private static String resolveUserIdFromUsername(AuthenticatedUser authorizedUser) {
+
+        String userId = null;
+        if (StringUtils.isNotBlank(authorizedUser.getTenantDomain()) &&
```

Comment:
Is authorizedUser not null always?

Reply:
It's taken from the `accessTokenDO` so i think it should not be null